syntax = "proto3";

package demo;

// Definición del servicio gRPC
service DemoService {
  // Método unario: recibe un mensaje y retorna una respuesta.
  rpc SendMessage(MessageRequest) returns (MessageResponse) {}

  // Método con respuesta en streaming: recibe un mensaje y retorna un stream de respuestas.
  rpc StreamMessages(MessageRequest) returns (stream MessageResponse) {}
}

// Mensaje de solicitud
message MessageRequest {
  string payload = 1;
  int32 id = 2;
}

// Mensaje de respuesta
message MessageResponse {
  string result = 1;
  int64 timestamp = 2;
}
